server/neptune/gateway/event/closed_pull_request_handler.go

```diff
@@ -0,0 +1,54 @@
+package event
+
+import (
+	"context"
+	"github.com/pkg/errors"
+	"github.com/runatlantis/atlantis/server/http"
+	"github.com/runatlantis/atlantis/server/logging"
+	"github.com/runatlantis/atlantis/server/lyft/feature"
+	"github.com/runatlantis/atlantis/server/neptune/gateway/pr"
+	"github.com/runatlantis/atlantis/server/neptune/workflows"
+)
+
+type prCloseSignaler interface {
+	SignalWorkflow(ctx context.Context, workflowID string, runID string, signalName string, arg interface{}) error
+}
+
+type ClosedPullRequestHandler struct {
+	WorkerProxy     workerProxy
+	Allocator       feature.Allocator
+	Logger          logging.Logger
+	PRCloseSignaler prCloseSignaler
+}
+
+func (c *ClosedPullRequestHandler) Handle(ctx context.Context, request *http.BufferedRequest, event PullRequest) error {
+	if err := c.WorkerProxy.Handle(ctx, request, event); err != nil {
+		c.Logger.ErrorContext(ctx, err.Error())
+	}
+
+	if err := c.handlePlatformMode(ctx, event); err != nil {
+		return errors.Wrap(err, "handling platform mode")
+	}
+	return nil
+}
+
+func (c *ClosedPullRequestHandler) handlePlatformMode(ctx context.Context, event PullRequest) error {
+	shouldAllocate, err := c.Allocator.ShouldAllocate(feature.LegacyDeprecation, feature.FeatureContext{
+		RepoName: event.Pull.HeadRepo.FullName,
+	})
+	if err != nil {
+		c.Logger.ErrorContext(ctx, "unable to allocate pr mode")
+		return nil
+	}
+	if !shouldAllocate {
+		c.Logger.InfoContext(ctx, "handler not configured for allocation")
+		return nil
+	}
+	return c.PRCloseSignaler.SignalWorkflow(
+		ctx,
+		pr.BuildPRWorkflowID(event.Pull.BaseRepo.FullName, event.Pull.Num),
```

Comment:
isn't this going to have the same problem we just talked about? ie. we receive closed events for all PRs and we need to know if a workflow exists for a given PR

Reply:
Yep, exactly. Forgot to add a comment here, but I was going to update it once we determine best way to check if a workflow exists

Reply:
filtering out serviceerror.NotFound

Reply:
we should add metrics around how often we're seeing this.

Reply:
Also worth thinking about how we will know whether service not found is something to ignore vs something we need to take action on.

Reply:
Added a metric, but I don't think we need to do much outside of of observability. Temporal confirms a call to a non-running workflow should return this error type, so we should ignore the underlying Github event. If there was ever a case we return this for the wrong reasons (a workflow does exist and is running), the workflow itself has a close event poller + full workflow timeout to shut itself down.